npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -708,16 +689,63 @@ def raise_resolvability_error(error_message, yarn_lock)
         ).returns(T::Boolean)
       end
       def pattern_in_message(patterns, message)
-        patterns.any? do |pattern|
+        patterns.each do |pattern|
           if pattern.is_a?(String)
-            return message.include?(pattern)
+            return true if message.include?(pattern)
           elsif pattern.is_a?(Regexp)
-            message = message.gsub(/\e\[[\d;]*[A-Za-z]/, "")
-            return message.match?(pattern)
+            return true if message.gsub(/\e\[[\d;]*[A-Za-z]/, "").match?(pattern)
           end
```

Comment:
Fixed the bug to go over all patterns instead of going over the first one when there is no match